src/base/bittorrent/session.cpp

```diff
@@ -1392,6 +1392,27 @@ void Session::banIP(const QString &ip)
     }
 }
 
+int Session::checkFlags(const QString &ip)
+{
+    libt::ip_filter filter = m_nativeSession->get_ip_filter();
+    boost::system::error_code ec;
+    libt::address addr = libt::address::from_string(ip.toLatin1().constData(), ec);
+    Q_ASSERT(!ec);
```

Comment:
Does this mean one can crash qBt just providing a malformed IP address string? Speaking more generally, in a function that is meant to check input for its validity you make an assertion that the input is valid.